giraffe/src/utils/PlotEnv.ts

```diff
@@ -149,22 +150,11 @@ export class PlotEnv {
     )
   }
 
-  public get yColumnType(): ColumnType {
-    //added to allow the y-axis type to be a string
-    //if there are multiple layers (ex. single stat + line graph),
-    //it will pick the yColumnType of the 1st layer (so BEWARE)
-    for (let i = 0; i < this.config.layers.length; i++) {
-      const layer: any = this.config.layers[i]
-      if (layer.yColumnType) {
-        return layer.yColumnType
-      }
-    }
-
-    return 'number'
-  }
-
-  public set yColumnType(columnType: ColumnType) {
-    this.config.yColumnType = columnType
+  public get yColumnLabels(): string[] {
+    return this.config.layers
+      .map((_, i) => this.getSpec(i))
+      .filter(spec => spec && spec.yColumnLabels)
+      .map(spec => spec.yColumnLabels)[0]
```

Comment:
let's clean this up a bit (the original algorithm was hacky and this is a bit more functional, but a bit unoptimized). I'm going to write out what I see in pseudocode and see if we can factor out the logic to be simpler and reflect what we're trying to do.

```
for every *config layer* in the *config layers collection*
  get the spec of this layer
  return the spec of this layer as an item in this array

for every *spec* in the *spec array* just created
  filter any item where spec is falsy or spec is truthy but its key *yColumnLabels* is falsy

for every *spec* in the *filtered spec array* just created
  return the *yColumnLabel* for this spec

get the 0th item of the array of *yColumnLabels*
  (which should be the first label we've encountered)
```
is that correct?

it seems like the intention is to get the first legitimate `yColumnLabel` from this collection and use that. Is that correct? 

If it's correct, can you think of a way to communicate that to someone reading this code without much context?

Reply:
here is the solution rashi & deniz came up with:

```
  public get yColumnLabels(): string[] {
    const currLayerSpec = this.config.layers.map((_, i) => this.getSpec(i))[0]

    if (currLayerSpec.type === 'mosaic') {
      return currLayerSpec.yColumnLabels
    }
    return []
  }
```